src/operations.jl

```diff
@@ -56,29 +79,83 @@ Base.:(==)(x::ArrayReferenceMeta, y) = false
 
 abstract type AbstractLoopOperation end
 
+"""
+`OperationType` is an `@enum` for classifying supported operations that can appear in
+`@avx` blocks. Type `LoopVectorization.OperationType` to see the different types.
+"""
 @enum OperationType begin
     constant
     memload
     compute
     memstore
     loopvalue
 end
+"An operation setting a variable to a constant value (e.g., `a = 0.0`)" constant
+"An operation setting a variable from a memory location (e.g., `a = A[i,j]`)" memload
+"An operation computing a new value from one or more variables (e.g., `a = b + c`)" compute
+"An operation storing a value to a memory location (e.g., `A[i,j] = a`)" memstore
+"??" loopvalue
```

Comment:
```julia
for i in 1:4, j in 0:3
    s += i * a[ i - j ]
end
```
In the operations
`i - j` and `i * ...`, `i` and `j` are represented as `loopvalue`s. LoopValues aren't needed for indexing, but are whenever they're a parent of a `compute` operation.